docs/DeveloperGuide.adoc

```diff
@@ -767,6 +825,132 @@ Use case ends.
 +
 Use case resumes at step 2.
 
+[discrete]
+=== Use case: Add person by prompt
+
+*MSS*
+
+1.  User requests to add persons without further details
+2.  AddressBook prompts user to input <info> of person to add
+3.  User inputs <info> as requested
+4.  AddressBook prompts (step 2 - 3) for the following information until all required information (r) has been filled:
+    1. name of person (r)
+    2. phone number (r)
+    3. email (r)
+    4. address (r)
+    5. tags
+    6. birthday
+5.  AddressBook adds person
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 5a. AddressBook shows error message if same person found
++
+Use case ends.
+
+[discrete]
+=== Use case: Upload image of Person from local files
+
+*MSS*
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+3.  User requests to upload image of a specific person in the list
+4.  AddressBook prompts for location of image
+5.  User inputs file path
+6.  AddressBook updates image
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 6a. The file path given is invalid.
++
+[none]
+** 6a1. AddressBook shows an error message.
++
+Use case resumes at step 4.
+
+* 6b. The file type of file given is invalid.
++
+[none]
+** 6b1. AddressBook shows an error message.
++
+Use case resumes at step 4.
+
+[discrete]
+=== Use case: Set user reference location
+
+*MSS*
+
+1.  User requests to set reference location
+2.  AddressBook updates reference location
++
+Use case ends.
+
+[discrete]
+=== Use case: Generate encoded contact details
+
+*MSS*
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+3.  User requests to select specific person
+4.  AddressBook prompts for which details to not encode
+5.  User inputs flags of details to not encode
+6.  AddressBook returns encoded string
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 6a. The flags given are invalid.
++
+[none]
+** 6a1. AddressBook shows an error message.
++
+Use case resumes at step 4.
+
+[discrete]
+=== Use case: Generate decoded contact details
+
+*MSS*
+
+1.  User requests to decode person details
+2.  AddressBook shows decoded details
+3.  AddressBook prompts user on whether to save contact
```

Comment:
Use Cases are the interactions between User and System.
So it should be a communication from User to System and System to User.
In this case perhaps line 2 and 3 should be combined together say `AddressBook shows decoded details and prompt user to save contact.`